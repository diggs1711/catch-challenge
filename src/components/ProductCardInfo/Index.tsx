import React from "react";

type TextSize = "small" | "large";

type ProductCardInfoProps = {
  text: string;
  size?: TextSize;
  strikethrough?: boolean;
};

const ProductCardInfo: React.FC<ProductCardInfoProps> = ({
  text,
  size = "small",
  strikethrough = false,
}) => {
  return (
    <div
      style={{
        fontSize: size === "small" ? "16px" : "24px",
        fontWeight: size === "small" ? "normal" : "bold",
        textDecoration: strikethrough ? "line-through" : "",
      }}
    >
      {text}
    </div>
  );
};

export default ProductCardInfo;
