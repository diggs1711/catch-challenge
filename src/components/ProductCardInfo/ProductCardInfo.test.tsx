import React from "react";
import { render } from "@testing-library/react";
import ProductCardInfo from "./Index";

describe("ProductCardInfo", () => {
  it("displays small text by default", () => {
    const { container } = render(<ProductCardInfo text="This is a test" />);
    expect(container).toMatchSnapshot();
  });

  it("displays large text", () => {
    const { container } = render(<ProductCardInfo text="This is a test" />);
    expect(container).toMatchSnapshot();
  });
});
