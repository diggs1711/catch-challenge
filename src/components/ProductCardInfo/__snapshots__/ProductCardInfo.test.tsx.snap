// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardInfo displays large text 1`] = `
<div>
  <div
    style="font-size: 16px; font-weight: normal;"
  >
    This is a test
  </div>
</div>
`;

exports[`ProductCardInfo displays small text by default 1`] = `
<div>
  <div
    style="font-size: 16px; font-weight: normal;"
  >
    This is a test
  </div>
</div>
`;

exports[`ProductCardInfo displays strikethrough 1`] = `
<div>
  <div
    style="font-size: 16px; font-weight: normal; text-decoration: line-through;"
  >
    This is a test
  </div>
</div>
`;
